python/MooseDocs/commands/init.py

```diff
@@ -0,0 +1,125 @@
+
+    sqa_parser = init_subparser.add_parser('sqa', help="Initialize SQA documentation.")
+    sqa_parser.add_argument('--app', required=True, type=str, help="Name of application to use in SQA template files.")
+    sqa_parser.add_argument('--category', default=None, type=str, help="Name of application to use in SQA configuration category name; if not provided it will be the lower case of the 'app' option.")
+
+def _write_file(filename, content):
+    """Helper for easy mock"""
+    with open(filename, 'w') as fid:
+        fid.write(content)
+
+def init_sqa_docs(app, category):
+    """Adds markdown files that load MOOSE SQA template files."""
+
+    # Loop through MOOSE SQA template files
+    for tname in glob.glob(os.path.join(MooseDocs.MOOSE_DIR, 'framework', 'doc', 'content', 'templates', 'sqa', 'app*.template')):
+        template_name = os.path.basename(tname)
+
+        # Application SQA document filename
+        fname = '{}{}'.format(category, template_name[3:-9])
+        if fname.endswith('index.md'):
+            fname = 'index.md'
+
+        filename = os.path.join(os.getcwd(), 'content', 'sqa', fname)
+
+        # Create the directory, if needed
+        os.makedirs(os.path.dirname(filename), exist_ok=True)
+
+        # Write the content to the output file
+        content = "!template load file=sqa/{} app={} category={}".format(template_name, app, category)
+        _write_file(filename, content)
+
+def init_sqa_config(app, category):
+    """Creates default file to load from MooseDocs configuration."""
+    template = os.path.join(os.path.dirname(__file__), 'sqa_app.yml.template')
+
+    with open(template, 'r') as fid:
+        content = fid.read()
+
+    repo = mooseutils.git_repo(MooseDocs.ROOT_DIR)
+    content = mooseutils.apply_template_arguments(content, repo=repo, category=category, app=app)
+
+    filename = os.path.join(os.getcwd(), 'sqa_{}.yml'.format(category))
+    _write_file(filename, content)
+
+def init_sqa_report(app, category):
+    """Creates default SQA report configuration file."""
+    template = os.path.join(os.path.dirname(__file__), 'sqa_reports.yml.template')
+
+    with open(template, 'r') as fid:
+        content = fid.read()
+
+    content = mooseutils.apply_template_arguments(content, category=category, app=app)
+    filename = os.path.join(os.getcwd(), 'sqa_reports.yml')
+    _write_file(filename, content)
+
+def update_config_with_sqa(filename, app, category):
+    """Modifies the config.yml to support SQA."""
+
+    with open(filename, 'r') as fid:
+        config = mooseutils.yaml_load(filename, include=False)
+
+    if 'MooseDocs.extensions.sqa' not in config['Extensions']:
+        config['Extensions']['MooseDocs.extensions.sqa'] = dict()
+
+    if 'MooseDocs.extensions.template' not in config['Extensions']:
+        config['Extensions']['MooseDocs.extensions.template'] = dict(active=True)
+
+    sqa = config['Extensions']['MooseDocs.extensions.sqa']
+    sqa['active'] = True
```

Comment:
Why can't you have `dict(active=True)` on line 91 as in line 94 to remove this?

Reply:
This extension might already be in the config, thus that dict() call might not occur.